src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -157,15 +191,37 @@ def get_security_updates(self):
         out = self.invoke_package_manager(cmd)
         security_packages, security_package_versions = self.extract_packages_and_versions(out)
 
-        self.composite_logger.log("Discovered " + str(len(security_packages)) + " 'security' package entries.")
-        return security_packages, security_package_versions
+        if self.__pro_client_prereq_met:
+            ubuntu_pro_client_security_updates_query_success, ubuntu_pro_client_security_packages, ubuntu_pro_client_security_package_versions = self.ubuntu_pro_client.get_security_updates()
+
+        self.composite_logger.log_debug("Get Security Updates : [DefaultSecurityPackagesCount={0}][UbuntuProClientQuerySuccess={1}][UbuntuProClientSecurityPackagesCount={2}]".format(len(security_packages), ubuntu_pro_client_security_updates_query_success, len(ubuntu_pro_client_security_packages)))
+
+        if ubuntu_pro_client_security_updates_query_success:
+            return ubuntu_pro_client_security_packages, ubuntu_pro_client_security_package_versions
+        else:
+            return security_packages, security_package_versions
```

Comment:
As we are returning security_packages only in case of ubuntu_pro_client_security_updates_query_success = False, should we run command to get security_packages only in case of ubuntu_pro_client_security_updates_query_success = False?

But in that case, we will not have log for len(security_packages) which is currently logged in Line 197.

Reply:
We are doing this just for debugging in case there is an mismatch. This is in-line with all the pro related changes i am doing.